File: Two Pointers/two_sum_ii_input_array_is_sorted.py
"""
https://leetcode.com/problems/two-sum-ii-input-array-is-sorted/description/
"""

class Solution: #Time complexity O(n), Space complexity O(1)
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        l = 0
        r = len(numbers) - 1

        while l <= r:
            total = numbers[l] + numbers[r]

            if total > target:
                r -= 1
            elif total < target:
                l += 1
            else:
                return [l + 1, r + 1]


class Solution: #Time complexity O(n^2), Space complexity O(1)
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        for i in range(len(numbers)):
            for j in range(i+1, len(numbers)):
                if numbers[i] + numbers[j] > target:
                    break
                elif numbers[i] + numbers[j] == target:
                    return [i + 1, j + 1]
        

        

File: Two Pointers/two_sum_ii_input_array_is_sorted.js
//https://leetcode.com/problems/two-sum-ii-input-array-is-sorted/description/

class Solution {
    /**
     * @param {number[]} numbers
     * @param {number} target
     * @return {number[]}
     Time complexity O(n), Space complexity O(1)
     */
    twoSum(numbers, target) {
        let l = 0
        let r = numbers.length - 1

        while (l <= r){
            let total = numbers[l] + numbers[r]

            if (total > target){
                r -= 1
            } else if ( total < target){
                l += 1
            } else {
                return [l + 1, r + 1]
            }
        }
    }
}
